src/EventSubscriber/SluggableSubscriber.php

```diff
@@ -99,21 +99,29 @@ private function generateUniqueSlugFor(SluggableInterface $sluggable): void
 
         $uniqueSlug = $slug;
 
+        $sluggableRepository = $this->entityManager->getRepository(get_class($sluggable));
```

Comment:
Repository should not depend on repository.

Better use only `SluggableRepositoryInterface` interface to get the needed service.

Reply:
I get the repository for the entity, because different types of entities can have different requirements on the slug. See my comment on "isSlugUnique"

Reply:
Will this work with https://github.com/KnpLabs/DoctrineBehaviors/pull/541#discussion_r439817970 ?

Reply:
The SluggableEventSubscriber has a [depdendency on SluggableRepositoryInterface](https://github.com/KnpLabs/DoctrineBehaviors/blob/dc2c92532196f9d96d77880bc72a8aed67771f7b/src/EventSubscriber/SluggableSubscriber.php#L35). This repository [is used as fallback ](https://github.com/KnpLabs/DoctrineBehaviors/blob/dc2c92532196f9d96d77880bc72a8aed67771f7b/src/EventSubscriber/SluggableSubscriber.php#L103)if the repository from the entity does not implement SluggableRepositoryInterface.